docs/sections/papers/deita.md

```diff
@@ -0,0 +1,530 @@
+This dataset is composed of [ShareGPT](https://huggingface.co/datasets/shibing624/sharegpt_gpt4)
+(58k instruction/response pairs),
+[UltraChat](https://huggingface.co/datasets/stingning/ultrachat) (105k
+instruction/response pairs) and
+[WizardLM](https://github.com/nlpxucan/WizardLM) (143k
+instruction/response pairs). It sums to more than 300k
+instruction/response pairs. We aim to reduce the final subsample to 6k
+instruction/response pairs.
+
+## 0. Setup the notebook and packages
+
+Let\'s prepare our dependencies
+
+```bash
+!pip install -qqq git+https://github.com/argilla-io/distilabel.git@asyncio-colab-fix#egg=distilabel[openai,hf-transformers]
+!pip install -qqq pynvml huggingface_hub argilla
+```
+
+We can access secrets in either Google colab or locally: 
+
+```python
+import os
+
+try:
+    from google.colab import userdata
+
+    os.environ["OPENAI_API_KEY"] = userdata.get("OPENAI_API_KEY")
+except ImportError:
+    !pip install -qqq python-dotenv
+    
+    from dotenv import load_dotenv
+    load_dotenv()
+```
+
+We will authenticate to the Hugging Face Hub to access the models.
+
+```python
+from huggingface_hub import notebook_login
+
+notebook_login()
+```
+```python
+Import distilabel 
+
+    from distilabel.llm.huggingface.transformers import TransformersLLM
+    from distilabel.llm.openai import OpenAILLM
+    from distilabel.pipeline.local import Pipeline
+    from distilabel.steps.conversation import ConversationTemplate
+    from distilabel.steps.deita import DeitaFiltering
+    from distilabel.steps.expand import ExpandColumns
+    from distilabel.steps.generators.huggingface import LoadHubDataset
+    from distilabel.steps.task.complexity_scorer import ComplexityScorer
+    from distilabel.steps.task.evol_instruct.base import EvolInstruct
+    from distilabel.steps.task.evol_quality.base import EvolQuality
+    from distilabel.steps.task.generate_embeddings import GenerateEmbeddings
+    from distilabel.steps.task.quality_scorer import QualityScorer
+```
+
+Define the distilabel Pipeline and load the dataset from the Hugging Face Hub.
+
+```python
+pipeline = Pipeline(name="DEITA")
+
+load_data = LoadHubDataset(
+        name="load_data", batch_size=100, output_mappings={"prompt": "instruction"}, pipeline=pipeline
+    )
+```
+
+## 1. EVOL-INSTRUCT: Generate Instructions with an LLM
+
+[Evol-Instruct](https://arxiv.org/abs/2304.12244) automates the creation
+of complex instruction data for training large language models (LLMs) by
+progressively rewriting an initial set of instructions into more complex
+forms. This generated data is then used to fine-tune a model named
+WizardLM. Evaluations show that instructions from Evol-Instruct are
+superior to human-created ones, and WizardLM achieves performance close
+to or exceeding GPT3.5-turbo in many skills.
+
+In distilabel 1.0, we initialise each step of the data generation
+pipeline. Later, we\'ll connect them together.
```

Comment:
Are the line-breaks ignored when building the docs? I'd say my opinionated view of this is to keep the paragraphs within the same line, but maybe there's a rationale under this one besides maybe readability?